### PR TITLE
Fix the transpose of linear restoring matrix to make roll mode rows to be 0

### PR DESCRIPTION
**Pull request by salhus**
image

The roll mode rows should be 0.
The readBEMIOh5 function uses the permute command to make matrix dimensions compatible to WEC-Sim. This essentially flips the matrix, such that, $X(1:nFreq, b, a)$ becomes $X(a, b, 1:nFreq)$. This works well due to matrix symmetry. However, the roll row for linear restoration matrix must be 0.
This follows from Issue #1028

### CODE REVIEW

**Comment by nathanmtom** [comment]
@salhus Thanks for looking into the issue and submitting a fix. I've verified that the additional line of code you have included will zero out the last row of the 6x6 hydrostatic restoring matrix. Which is consistent that the yaw-roll and yaw-pitch coupling terms in the 6,4 and 6,5 indices are zero based on the WAMIT documentation you have cited. Please review my one comment above on how you are handling the matrix indices as my recommendation seems more direct, but I might not understand the reason behind that implementation. After we resolve that minor comment this would be fine to merge.

File: source/functions/BEMIO/readBEMIOH5.m [hunk 1]
```diff
@@ -53,7 +53,7 @@
 
 % Read hydrostatic stiffness
 hydroData.hydro_coeffs.linear_restoring_stiffness = reverseDimensionOrder(h5read(filename, [h5BodyName '/hydro_coeffs/linear_restoring_stiffness']));
-
+hydroData.hydro_coeffs.linear_restoring_stiffness(6:6:end,:) = 0*hydroData.hydro_coeffs.linear_restoring_stiffness(6:6:end,:);
```
[the remark this answers is not in the document]
Comment (salhus) [reply source/functions/BEMIO/readBEMIOH5.m]: Hi @nathanmtom 
I left that as a safe-guard thinking if the matrix was $6N\times6N$.
I just made a commit that only does that to the $(6,:)$

[the remark this answers is not in the document]
Comment (nathanmtom) [reply source/functions/BEMIO/readBEMIOH5.m]: @salhus Thanks for the clarification, I see why you chose to implement the prior option; however, I do believe that when this function is called it requires the body number and should only have a 6x6 hydrostatic matrix upon import. If you can double check this, I think sticking with the (6,:) is easier to read, but if we are indeed getting multiple bodies read in then we should stick with your implementation.

[the remark this answers is not in the document]
Comment (salhus) [reply source/functions/BEMIO/readBEMIOH5.m]: @nathanmtom 
Thanks for pointing this out. The imports do happen on a body by body basis, so the $(6,0)$ elements being zeroed makes more sense.

**Comment by nathanmtom** [comment]
Thanks @salhus for confirming that the function imports body hydrostatics one at a time. I am happy with the implementation and will merge this PR.

**Comment by salhus** [comment]
@nathanmtom Thanks Nathan.

**Pull request merged**